### PR TITLE
More regs

### PR DESCRIPTION
**Pull request by AGSaidi**

### CODE REVIEW

**Comment by andre-richter** [comment]
Thanks! I’ll be able to look at this and publish in CW 37.

File: src/registers/vtcr_el2.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        /// walks using `VTTBR_EL2` and `VSTTBR_EL2`
+        SH0 OFFSET(12) NUMBITS(2) [
+            /// Non-shareable
+            Non = 0b00,
+            /// Outer sharable
+            Outer = 0b10,
+            /// Inner sharable
+            Inner = 0b11,
+        ],
+        /// Outer cacehability attribute for memory associated with translation table
```
Comment (nchong-at-aws) [inline src/registers/vtcr_el2.rs]:
```suggestion
        /// Outer cacheability attribute for memory associated with translation table
```

and below too

File: src/registers/vtcr_el2.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        /// Granule size used for `VTTBR_EL2`
+        TG0 OFFSET(14) NUMBITS(2) [
+            /// Granule size of 4KB
+            Granule4KB = 0b00,
+            /// Granule size of 16KB
+            Granule16KB = 0b10,
+            /// Granule size of 64KB
+            Granule64KB = 0b01,
+        ],
+        /// Sharability attribute for memory associated with translation table
```
Comment (nchong-at-aws) [inline src/registers/vtcr_el2.rs]:
```suggestion
        /// Shareability attribute for memory associated with translation table
```

**Comment by andre-richter** [comment]
@nchong-at-aws did your review cover the bitfields of the reg definitions?

File: src/registers/vtcr_el2.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            /// 40 bits, 1TB
+            PA1TB = 0b010,
+            /// 42 bits, 4TB
+            PA4TB = 0b011,
+            /// 44 bits, 16TB
+            PA16TB = 0b100,
+            /// 48 bits, 256TB
+            PA256TB = 0b101,
+            /// 52 bits, 4PB
+            PA4PB = 0b110,
```
Comment (KarimAllah) [inline src/registers/vtcr_el2.rs]: I would have preferred if the number of bits is also encoded in the field name e.g PA_32BIT_4G
What do you think?

**Review by KarimAllah — APPROVED** [review]
Reviewed bitfields too.

File: src/registers/tcr_el2.rs [hunk 4]
```diff
@@ -18,6 +18,20 @@ use tock_registers::{
 register_bitfields! {u64,
     pub TCR_EL2 [
 
+        /// When FEAT_HAFDBS is implemented hardware can update the dirty flags in the stage1
+        /// descriptors
+        HD OFFSET(21) NUMBITS(1) [
```
Comment (KarimAllah) [inline src/registers/tcr_el2.rs]: s/21/22 here

File: src/registers/id_aa64mmfr1_el1.rs [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        /// Speculative reads can generate SError Interrupts
+        SpecSEI OFFSET(24) NUMBITS(4) [
+            /// PE never generates SError interrupts on a speculative read
+            Never = 0b0000,
+            /// PE may generate SError interrupts on a speculative read
+            Maybe = 0b0001
+        ],
+
+        /// Privileged Access Never support
+        PAN OFFSET(16) NUMBITS(4) [
```
Comment (KarimAllah) [inline src/registers/id_aa64mmfr1_el1.rs]: s/16/20

File: src/registers/vtcr_el2.rs (hunk 3 above)
Comment (andre-richter) [reply src/registers/vtcr_el2.rs]: Good suggestion imo 👍

**Comment by andre-richter** [comment]
Thanks!

**Comment by andre-richter** [comment]
Included in https://crates.io/crates/cortex-a/7.5.0

File: src/registers/id_aa64mmfr1_el1.rs (hunk 5 above)
Comment (AGSaidi) [reply src/registers/id_aa64mmfr1_el1.rs]: Thanks Karim. I don't know how i messed these up.

**Pull request merged**